CONTRIBUTING.md

```diff
@@ -0,0 +1,22 @@
+- If anything isn't clear in the code, feel free to create an
+  [issue on GitHub][GitHubIssues] or [on GitLab][GitLabIssues] with a question.
+
+- If you want to add support for a hosting provider which is not yet supported:
+
+  1. Before making code changes, please create an [issue on GitHub][GitHubIssues]
+     or [on GitLab][GitLabIssues] to track the ongoing work.
+  2. Follow instructions at the top of class `GitHosting`.  Use existing
+     subclasses of class `GitHosting` for how various methods of class
+     `GitHosting` can be overridden.
+
+- Please keep the project's [goals and non-goals](./README.md#goals) in mind.
+
+- If your pull request requires changes in class `GitHosting` or top-level
+  functions, please be aware that ensuring that the userscript still works on
+  supported Git hostings is the top priority.  This may take a while to do due
+  to limited access to some of the supported hostings.
+
+- Before merging, your commits might be reorganized.
```

Comment:
This may be not clear enough, but I suppose you can cross that bridge when you get there :)